src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -2604,6 +2573,205 @@ class C : I
 }
 ";
         var comp = CreateCompilationWithRequiredMembers(code);
-        comp.VerifyDiagnostics();
+        comp.VerifyDiagnostics(
+            // (5,9): error CS9506: Required member 'C.P' must be set in the object initializer or attribute constructor.
+            // _ = new I() { P = 1 };
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "I").WithArguments("C.P").WithLocation(5, 9)
+        );
+    }
+
+    [Fact]
+    public void RequiredMemberInAttribute_NotSet()
+    {
+        var code = @"
+using System;
+
+[Attr]
```

Comment:
Consider updating the feature proposal to clearly identify this scenario as being supported by the feature.